### ISSUE TITLE
Setup Project/Component creation

### ISSUE THREAD

**Issue by rbclark** [mitre/vulcan#95]
The Alpha Vulcan had a concept of projects. In new Vulcan we are aiming to have a concept of Components and Projects: #58. In order to do this we need to setup the ability to create Components in the new version, and then port the remaining code required for regular project work as well.

**Comment by Bialogs** [comment]
Instead of having two different tables can we consider just having them all be projects and then only calling a project a component when it is linked to another project?

**Comment by jkufro** [comment]
> Instead of having two different tables can we consider just having them all be projects and then only calling a project a component when it is linked to another project?

Are projects able to “have many rules” just like components will “have many rules”

**Comment by Bialogs** [comment]
Yes in my mind there is just a semantic difference between component and project. A project is only a "project" when viewed from the correct perspective.

```
A (project)
 |-- B (component)
       | -- Rule 
|-- D (component)
      | -- E (component)
      | -- F (component)

C (project)
 |-- Rule 
```

**Comment by Bialogs** [comment]
Project Creation Fields
* Select the SRG that the project will be based on
* Name it
* Determine Rule Title Prefix
